### ISSUE TITLE
Is there any way to launch specific programs without decorations?

### ISSUE THREAD

**Issue by bcd0101** [JLErvin/berry#80]
This is because I'm using **st** with **berry** and **picom** and I can't get opacity with decorations enabled.

**Comment by bcd0101** [comment]
Or I'm making a stupid question?

**Comment by JLErvin** [comment]
There are no stupid questions here. 
So, first off I've been playing around with deocration refactoring that would allow this. Sadly `berry`'s implementation of double borders + title bars make this difficult - although it is something I am working on.
In the meantime, a pure _unix_ approach might be to use something like [ruler](https://github.com/tudurom/ruler) which executes a set of commands when a window of a certain type is opened. You could use this to toggle the `decorate_new` flag on `berry`.

**Comment by bcd0101** [comment]
I've testing ruler with berry, but unfortunately it's not working as I expected. Of course ruler allow me to run a command after certain type of window appears in to the system.

> berryc decorate_new false

Is executed after open st turning off decorations for all new windows.

**Comment by bcd0101** [comment]
@JLErvin

This issue it's not too big for me, so if you say that you are working with decoration refactoring I only have to be thankful. 

I'll waiting for updates.

**Comment by superstefan420** [comment]
> I've testing ruler with berry, but unfortunately it's not working as I expected. Of course ruler allow me to run a command after certain type of window appears in to the system.
> 
> > berryc decorate_new false
> 
> Is executed after I open st, turning off decorations for all new windows.

it works great if you use toggle_decorations instead

**Comment by bcd0101** [comment]
@superstefan420 I don't think so, because toogle decorations will turn off all window decorations, but, I would like to launch only one without decorations.

Similarly could happen with a defined size of a specific window.

**Comment by JLErvin** [comment]
hi @superstefan420 , I took at shot at adding this a while ago in https://github.com/JLErvin/berry/tree/blackberry. `berryc` would work well with this so long as there was support for toggle decorations for a specific window (specified by an id) rather than the actively focused window. 

Unfortunately I haven't had much time to work on `berry` lately, but I can try to make this a relative priority in the near future since I know this is a requested feature. I'm always open to reviewing PRs as well if people want to take a shot at it themselves.

**Comment by superstefan420** [comment]
> @superstefan420 I don't think so, because toogle decorations will turn off all window decorations, but, I would like to launch only one without decorations.
> 
> Similarly could happen with a defined size of a specific window.

I’m telling you it works lol.. it has the same effect as binding it with sxhkd it only effects the focused window

**Comment by superstefan420** [comment]
> hi @superstefan420 , I took at shot at adding this a while ago in https://github.com/JLErvin/berry/tree/blackberry. `berryc` would work well with this so long as there was support for toggle decorations for a specific window (specified by an id) rather than the actively focused window.
> 
> Unfortunately I haven't had much time to work on `berry` lately, but I can try to make this a relative priority in the near future since I know this is a requested feature. I'm always open to reviewing PRs as well if people want to take a shot at it themselves.

Honestly i like how berry is right now and I think using something like ruler for window rules fits perfectly with the berry philosophy .

I do like how bspwm handles rules though so you can look at that if you want to implement it right into berry
